jetty-security/src/main/java/org/eclipse/jetty/security/authentication/SpnegoAuthenticator.java

```diff
@@ -72,8 +72,12 @@ public Authentication validateRequest(ServletRequest request, ServletResponse re
             return new DeferredAuthentication(this);
         }
 
+        final boolean hasNegotiateHeader = header == null ? false : header.startsWith(HttpHeader.NEGOTIATE.asString();
         // check to see if we have authorization headers required to continue
-        if ( header == null )
+        // A challenge should be sent if:
+        //   1. There was not Authorization header provided
+        //   2. There was an Authorization header for a type other than Negotiate
+        if (header == null || (header != null && !hasNegotiateHeader))
```

Comment:
this can just be !hasNegotiateHeader, as if  the header is null then line 75 sets hasNegotiateHeader to false.
Probably best to switch this if around then (else block first and remove the !).
Probably don't need the variable then either.